ssl/src/main/java/org/wildfly/security/ssl/SSLParametersUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.ssl;
+
+import com.google.gson.ExclusionStrategy;
+import com.google.gson.FieldAttributes;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import javax.net.ssl.SSLParameters;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+class SSLParametersUtil {
+
+    /**
+     * Uses Gson to copy unknown SSLParameters' fields added in java 9 or later. Setter methods are used for fields available in Java 8.
+     *
+     * @param original SSLParameters that should be applied to new instance
+     * @return instance of SSLParameters with fields copied from original
+     */
+    static SSLParameters copySSLParameters(SSLParameters original) {
```

Comment:
@fjuma I cannot use setters for some fields  ( `maximumPacketSize`, `applicationProtocols`, `enableRetransmissions`) because these fields have been introduced in Java 9. To set these fields I used gson - but this has disadvantage of being slower. I think setSSLParameters method might be used by many threads? Which means this might create bottleneck when running java 9 and above. Let me know whether gson is alright, or whether I should find faster library which does cloning through reflection and not string, or whether I should use reflection on my own, or wether I should ignore these parameters altogether - which would not set SSLParameters properly when running on newer java.

Reply:
edited last post ^